src/pages/en/guides/typescript.md

```diff
@@ -7,11 +7,14 @@ i18nReady: true
 
 Astro ships with built-in support for [TypeScript](https://www.typescriptlang.org/). You can import `.ts` and `.tsx` files in your Astro project, write TypeScript code directly inside your [Astro component](/en/core-concepts/astro-components/#the-component-script), and even use an [`astro.config.ts`](/en/guides/configuring-astro/#the-astro-config-file) file if you like.
 
-Astro doesn't perform any type checking itself. Type checking should be taken care of outside of Astro, either by your IDE or through a separate script. The [Astro VSCode Extension](/en/editor-setup/) automatically provides TypeScript hints and errors in your open files.
+Using TypeScript, you can prevent errors at runtime by defining the shapes of objects and components in your code. For example, if you use TypeScript to [type your component's props](#component-props), you'll get an error in your editor if you set a prop that your component doesn't accept. 
 
-## Setup
+You don't need to write TypeScript code in your Astro projects to benefit from it. The [Astro VSCode Extension](/en/editor-setup/) uses TypeScript to provide autocompletion, hints, and errors in your editor.
+
+The Astro dev server won't perform any type checking, but you can use a [separate script](#type-checking) to check for type errors from the command line.
 
-Astro starter projects include a `tsconfig.json` file in your project. Even if you don't write TypeScript code, this file is important so that tools like Astro and VS Code know how to understand your project. Some features (like npm package imports) aren't fully supported in the editor without a `tsconfig.json` file. If you install Astro manually, be sure to create this file yourself.
+## Setup
+Astro always treats your component code as TypeScript, but some features (like npm package imports) aren't fully supported in the editor without a `tsconfig.json` file. Even if you don't write TypeScript code, this file is important so that tools like Astro and VS Code know how to understand your project. Astro starter projects include this file, but if you install Astro manually, be sure to create it yourself.
```

Comment:
```suggestion
Astro starter projects include a `tsconfig.json` file in your project. Even if you don't write TypeScript code, this file is important so that tools like Astro and VS Code know how to understand your project. Some features (like npm package imports) aren't fully supported in the editor without a `tsconfig.json` file. If you install Astro manually, be sure to create this file yourself.
```
I'm not sold on this change at all. (Above suggestion reverts to original.) 

The "but" doesn't seem to be the right relationship between these two statements here. I also prefer starting out by mentioning the `tsconfig.json` file; seems to be more direct, especially when the heading is "Setup."  (Here's the setup file!)

- You'll see this file
- Even if you don't think you need it, you do
- if you installed manually, make it yourself.

I also don't immediately see the best place to stick in "Astro always treats your component code as TypeScript", which is the new idea you wanted to get in there. Maybe that's a clue that it doesn't belong in the "setup" section? What about throwing this idea up in the "you'll get benefits even if you don't use it" section?

Reply:
Agreed, I think I overthought this one!